packages/relay/src/lib/services/ethService/ethFilterService/index.ts

```diff
@@ -137,4 +142,47 @@ export class FilterService implements IFilterService {
     this.logger.trace(`${requestIdPrefix} newPendingTransactionFilter()`);
     return predefined.UNSUPPORTED_METHOD;
   }
+
+  public async getFilterChanges(filterId: string, requestIdPrefix?: string): Promise<string[] | Log[] | JsonRpcError> {
+    this.logger.trace(`${requestIdPrefix} getFilterChanges(${filterId})`);
+    FilterService.requireFiltersEnabled();
+
+    const cacheKey = `${constants.CACHE_KEY.FILTERID}_${filterId}`;
+    const filter = this.cache.get(cacheKey, this.ethGetFilterChanges, requestIdPrefix);
+
+    if (!filter || this.supportedTypes.indexOf(filter.type) === -1) {
```

Comment:
> this.supportedTypes.indexOf(filter.type) === -1

should not this be validated when the filter is saved into the cache?

Reply:
It's not possible to create a filter with an unsupported type, but I think it's good to have this check just in case

Reply:
I see your point, but not sure to agree. if is really not needed it only makes the code harder to read (readability impact). and a very small unneeded computation cost.

I would leave this comment here open in case someone else has his opinion on this.

For me is just a Nit and would not matter too much if we have it or not.